src/Akeneo/Category/back/Infrastructure/Controller/InternalApi/UpdateAttributeController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\Infrastructure\Controller\InternalApi;
+
+use Akeneo\Category\Api\Command\CommandMessageBus;
+use Akeneo\Category\Api\Command\Exceptions\ViolationsException;
+use Akeneo\Category\Application\Command\UpdateAttributeCommand\UpdateAttributeCommand;
+use Oro\Bundle\SecurityBundle\SecurityFacade;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class UpdateAttributeController
+{
+    public function __construct(
+        private readonly SecurityFacade $securityFacade,
+        private readonly CommandMessageBus $categoryCommandBus,
+        private readonly NormalizerInterface $constraintViolationNormalizer,
+    ) {
+    }
+
+    /**
+     * @param string $attributeUuid We doesn't use the $templateUuid in the code, but we keep it for interface convention. It maintains explicit link between the attribute UUID with its template.
+     */
+    public function __invoke(Request $request, string $templateUuid, string $attributeUuid): Response
+    {
+        if (!$this->securityFacade->isGranted('pim_enrich_product_category_template')) {
+            throw new AccessDeniedException();
+        }
+
+        $data = $request->toArray();
+
+        try {
+            $command = UpdateAttributeCommand::create(
+                attributeUuid: $attributeUuid,
+                isRichTextArea: $data['isRichRextArea'],
+            );
+            $this->categoryCommandBus->dispatch($command);
+        } catch (ViolationsException $violationsException) {
+            $normalizedViolations = [];
+            foreach ($violationsException->violations() as $violation) {
+                $normalizedViolations[] = $this->constraintViolationNormalizer->normalize(
+                    $violation,
+                    'internal_api',
+                    ['attribute' => $attributeUuid],
+                );
+            }
+
+            return new JsonResponse(['values' => $normalizedViolations], Response::HTTP_BAD_REQUEST);
+        } catch (\InvalidArgumentException $invalidArgumentException) {
+            throw new NotFoundHttpException($invalidArgumentException->getMessage());
+        }
```

Comment:
```suggestion
        } catch (\InvalidArgumentException $invalidArgumentException) {
            throw new NotFoundHttpException($invalidArgumentException->getMessage());
        }
```
I think that it's too implicit to transform a `InvalidArgumentException` into a `NotFoundHttpException`. `InvalidArgumentException` can be triggered from many different context / classes. Either we need a custom domain exception like `AttributeNotFoundException` that can be catched reliably OR (what I will suggest) don't catch anything, we will get a 500 instead of a 404 on the POST endpoint but the frontend will fail in both situation. Wdyt?